src/openforms/registrations/contrib/objects_api/plugin.py

```diff
@@ -144,6 +148,13 @@ def register_submission(
         context = {
             "_submission": submission,
             "productaanvraag_type": options["productaanvraag_type"],
+            "payment": {
+                "completed": submission.payment_user_has_paid,
+                "amount": submission.payments.sum_amount(),
```

Comment:
I think we need to properly stringify this before passing to the template engine, since the localization in the templates will cause problems with large numbers. Please add a test case for that situation too (try using number `1_000` but also `3.141592` and see how many decimals are outputted and what the rounding behaviour is).

Because floats are inherently limited, I think the resulting JSON type should be a string where a `.` is used as decimal separator and no grouping occurs, so expected is:

* `1_000` -> `"1000"`
* `3.1415926535` -> `"3.1415926535"`

Unfortunately I don't see a [`format`](https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-validation-00#page-13) for decimal, so maybe a string isn't the correct solution as in theory the object type schema can also specify the value to have a particular precision by using:

```
type: number
multipleOf: 0.01
```

So perhaps accepting the limitations of floats is okay here, since we process the payment already anyway :thinking:

Reply:
Discussed:
The object type schema should dictate the precision of this field (`type: number, multipleOf: 0.01`). We then send a number in the json and not a string.